src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure implements Drawable {
+    private double width;
+    private double height;
+    private double diagonal;
+
+    public Rectangle(String color, double width,
+                     double height) {
+        super(color);
+        this.width = width;
+        this.height = height;
+        this.diagonal = Math.sqrt(width * width + height * height);;
```

Comment:
There is an extra semicolon at the end of the line after calculating the diagonal. This is not a critical issue, but it is unnecessary and can be removed for cleaner code.